Sources/SQLiteKit/SQLiteConnectionSource.swift

```diff
@@ -6,24 +7,26 @@ import NIOCore
 
 public struct SQLiteConnectionSource: ConnectionPoolSource {
     private let configuration: SQLiteConfiguration
+    private let actualPath: URL
     private let threadPool: NIOThreadPool
 
     private var connectionStorage: SQLiteConnection.Storage {
-        switch self.configuration.storage {
-        case .memory(let identifier):
-            return .file(
-                path: "file:\(identifier)?mode=memory&cache=shared"
-            )
-        case .file(let path):
-            return .file(path: path)
-        }
+        .file(path: self.actualPath.absoluteString)
     }
     
     public init(
         configuration: SQLiteConfiguration,
         threadPool: NIOThreadPool
     ) {
         self.configuration = configuration
+        switch configuration.storage {
+        case .memory(identifier: let identifier):
+            let filenameSafeIdentifer = String(identifier.map { $0 == ":" ? "_" : ($0 == "\\" ? "+" : ($0 == "/" ? "-" : $0)) })
```

Comment:
emmmmmm can't say i like this ....... it could possibly be a dictionary lookup like:
```swift
let charMap: [Character: Character] = [
    ":": "_",
    "\\": "+",
    "/": "-",
]
Then `identifier.map { charMap[$0] ?? $0 }`
```
I'm also guessing Swift itself has some "character encoding stuff" for file names?

Reply:
I redid it, 'cause you were right 🙂.